urllib3/contrib/pyopenssl.py

```diff
@@ -138,7 +138,19 @@ def _dnsname_to_stdlib(name):
     then on Python 3 we also need to convert to unicode via UTF-8 (the stdlib
     uses PyUnicode_FromStringAndSize on it, which decodes via UTF-8).
     """
-    name = idna.encode(name)
+    def idna_encode(name):
```

Comment:
I'm not about to block this PR on this, but it does worry me that our pyopenssl integration has a very low bus factor. Tucking this in here doesn't seem entirely conducive to long-term maintenance but it's only used here so it's not presently very harmful either.